pkg/component/autoscaling/vpa/crd.go

```diff
@@ -71,3 +83,13 @@ func (v *vpaCRD) Destroy(ctx context.Context) error {
 
 	return nil
 }
+
+// Wait signals whether a CRD is ready or needs more time to be deployed.
+func (v *vpaCRD) Wait(ctx context.Context) error {
+	return kubernetesutils.WaitUntilCRDManifestsReady(ctx, v.client, resourceNames)
+}
+
+// WaitCleanup for destruction to finish and component to be fully removed. crdDeployer does not need to wait for cleanup.
+func (v *vpaCRD) WaitCleanup(_ context.Context) error {
+	return nil
+}
```

Comment:
It's not ideal that we don't use the CRDDeployer just because the `WaitCleanup` is different. You could for example work around by overloading `WaitCleanup` or by using [component.OpDestroyWithoutWait](https://github.com/gardener/gardener/blob/4d1d22130d77fa8f1391efd807ec205964482a73/pkg/component/deploy.go#L39).

Reply:
Like we talked about, I will use the generic `CRDDeployer` as an anonymous field for the individual deployers.